src/ui.jsx

```diff
@@ -0,0 +1,392 @@
+    componentDidMount() {
+        this._initialiseFileListener();
+    }
+
+    /**
+     * Sets up a listener so that we can read in robot instructions from a suitable file.
+     */
+    _initialiseFileListener() {
+
+        /*eslint-disable no-alert*/
+
+        // Check for File APIs support - Taken from HTML5Rocks.com
+        if (window.File && window.FileReader && window.FileList && window.Blob) {
+            // File APIs supported
+        } else {
+            alert("File APIs not fully supported in this browser, loading instructions from a file is disabled.");
+            return;
+        }
+
+        var reader = new FileReader();
+        var fileList = document.getElementById("fileInput");
+        var editor = document.getElementById("editor");
+
+        fileList.addEventListener("change", function() {
+
+            var file = fileList.files[0];
+
+            reader.onload = function() {
+
+                // Check the MIME type of the file to see if it's a text file
+                if (file.type.match("text/*")) {
+                    editor.value = reader.result;
+                } else {
+                    alert("File extension not supported!");
+                }
+
+            };
+
+            reader.readAsText(file);
+
+        });
+
+    }
+
+    /**
+     * Save the instructions entered into the editor into the application's state when the submit button is clicked.
+     */
+    _handleGoClick() {
+        var editor = document.getElementById("editor");
+        this.props._setup(editor.value);
+    }
+
+    _handleSkipClick() {
+        // TODO
+    }
+
+    /**
+     * [render description]
+     * @return {[type]} [description]
+     */
+    render() {
+
+        return (
+
+            <div id="sidebar">
+
+                <textarea
+                    id="editor" placeholder="Enter your code here!" rows="30" cols="75">
+                </textarea>
+
+                <div id="buttonBox">
+
+                    <input className="buttons" type="file" id="fileInput" name="file" />
+
+                    <button className="buttons" id="goButton" type="button"
+                        onClick={this._handleGoClick.bind(this)}>
+                            <div className="buttonText">Go!</div>
+                    </button>
+
+                    <button className="buttons" id="skipButton" ref="skipButton" type="button">
+                        <div className="buttonText">Skip Animation</div>
+                    </button>
+
+                </div>
+
+            </div>
+
+        );
+
+    }
+
+}
+
+/**
+ * Used to contain output components that display output and error messages to the user.
+ */
+class OutputBox extends React.Component {
+
+    constructor() {
+
+        super();
+
+        // Contains an value that can be incremented to stop React complaining about a lack of key
+        this.state = {
+            id: -1
+        };
```

Comment:
Remember we discussed this:

```js
this.props.outputData.map((item, index) => <Whatever key=index />)
```

Rather than using a counter in state. It isn't really state at all.